### ISSUE TITLE
net/frr: Multiple BGP peers causes config generation error

### ISSUE THREAD

**Issue by g-a-c** [opnsense/plugins#2623]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

[X] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md

[X] I have searched the existing issues and I'm convinced that mine is new.

[X] The title contains the plugin to which this issue belongs

**Describe the bug**
Following the upgrade to 21.7.4, having multiple BGP peers present (even if some are disabled) causes an error when the configuration is saved:

```
Oct 30 08:27:36 opnsense.home.arpa configd.py[57229]: generate template container OPNsense/Quagga
Oct 30 08:27:37 opnsense.home.arpa configd.py[57229]: [16ef5067-e51e-459e-81c5-a68595dba55e] Inline action failed with OPNsense/Quagga OPNsense/Quagga/bgpd.conf 'list object' has no attribute 'address' at Traceback (most recent call last):   File "/usr/local/opnsense/service/modules/template.py", line 264, in _generate     content = j2_page.render(cnf_data)   File "/usr/local/lib/python3.8/site-packages/jinja2/environment.py", line 1304, in render     self.environment.handle_exception()   File "/usr/local/lib/python3.8/site-packages/jinja2/environment.py", line 925, in handle_exception     raise rewrite_traceback_stack(source=source)   File "/usr/local/opnsense/service/modules/../templates/OPNsense/Quagga/bgpd.conf", line 64, in top-level template code     {%   if helpers.exists('OPNsense.quagga.bgp.neighbors.neighbor') and ':' not in OPNsense.quagga.bgp.neighbors.neighbor.address %} jinja2.exceptions.UndefinedError: 'list object' has no attribute 'address'  During handling of the above exception, another exception occurred:  Traceback (most recent call last):   File "/usr/local/opnsense/service/modules/processhandler.py", line 506, in execute     return ph_inline_actions.execute(self, inline_act_parameters)   File "/usr/local/opnsense/service/modules/ph_inline_actions.py", line 51, in execute     filenames = tmpl.generate(parameters)   File "/usr/local/opnsense/service/modules/template.py", line 341, in generate     raise render_exception   File "/usr/local/opnsense/service/modules/template.py", line 332, in generate     for filename in self._generate(template_name, create_directory):   File "/usr/local/opnsense/service/modules/template.py", line 267, in _generate     raise Exception("%s %s %s" % (module_name, template_filename, render_exception)) Exception: OPNsense/Quagga OPNsense/Quagga/bgpd.conf 'list object' has no attribute 'address'
Oct 30 08:27:37 opnsense.home.arpa configd.py[57229]: [2ac77799-3cee-444a-ad32-80e3df6365d8] starting quagga
```

**To Reproduce**
1. Install OPNsense 21.7.4
2. Install `os-frr`
3. Configure a basic BGP setup with 1 neighbour
4. Observe that the configuration in `/usr/local/etc/frr/bgpd.conf` is generated correctly
5. Add a second peer and save the configuration
6. Observe that the configuration in `/usr/local/etc/frr/bgpd.conf` has not been regenerated and is missing the second peer
7. Observe the above error in the `configd` log file

**Expected behavior**
The second peer is added to the configuration correctly

**Screenshots**

**Relevant log files**
If applicable, information from log files supporting your claim.

**Additional context**

**Environment**
Software version used and hardware type if relevant.
e.g.:

OPNsense 21.7.4 amd64

**Comment by winglessza** [comment]
Confirmed. Can reproduce on new install.

**Comment by mimugmail** [comment]
@AdSchellevis @fichtner did you change something in Core/Base? This templating error also occurs on 1.22 which is out since 21.7 and I can't remember an execute error ever since. Also 1.23 in devel is affected and I always test the installation, can't imagine this is somthing inside frr.

**Comment by fichtner** [comment]
Jina2 update to version 3 maybe?

**Comment by AdSchellevis** [comment]
highly unlikely,  more likely the error is as propogated:

https://github.com/opnsense/plugins/blob/eebe9b346ff9667b588e8aaaddfbfecfeb02a913/net/frr/src/opnsense/service/templates/OPNsense/Quagga/bgpd.conf#L64

(note the address, which according to the next line is in a list container)

**Comment by g-a-c** [comment]
I think @AdSchellevis is right and I think I see the issue so I'd like to take a look at this as I have some PRs open for other FRR issues I've found (#2602, #2604, #2606) so it may be that I can tie this fix into one of those

**Comment by mimugmail** [comment]
Great, looking forward to review :)

**Comment by g-a-c** [comment]
So, sanity check me here...

I noticed there was a lot of duplicated code/templating due to there being both `ipv4` and `ipv6` address families. So I took it upon myself to build on the PR I opened for #2606 and made the entire `address-family` block a loop in order to reduce the duplication. The options are the same in the web UI no matter which type of peer you're configuring so this seems safe to me. However this means that it's quite a major refactor which kind of renders the other PRs moot because there are conflicting changes.

How would people feel about me combining my other PRs with the work I did for this, rebasing everything on `master` and submitting a single combined PR to be reviewed to close all four issues?
